src/framework.rs

```diff
@@ -71,9 +69,11 @@ pub fn run_bot<B: Bot>(mut bot: B) -> Result<(), Box<Error>> {
     let mut packets = rlbot.packeteer();
 
     loop {
-        let packet = packets.next()?;
-        let input = bot.tick(&packet);
-        rlbot.update_player_input(input, player_index)?;
+        let packet = packets.next_flatbuffer()?;
+        if packet.gameInfo().unwrap().isRoundActive() {
```

Comment:
Mentioned in a comment earlier, it's because we get "bad" packets, ie packets missing basic things like the ball info, which would then require the user to add their own checks. If a user wants those, they can always use the lower level interface, but since this is a medium-level interface and seemed like a lame thing to have to worry about as a bot maker, I just filtered here.

If you feel strongly I can remove.

Reply:
These bad packets don't occur with the ffi interface. You mentioned ffi uses flatbuffer internally. What's the difference? Does it just keep a persistent buffer and skip copying the missing fields? I'll have to look at their code.

Whatever the solution, I don't think celebrating should be limited to the low level interface, since that means you can't celebrate after a tournament win, and that is the best time to celebrate 😄

Reply:
Yes, that is exactly what happens, [it uses a buffer](https://github.com/RLBot/RLBot/blob/82892ff093c22cbf25a8f8542de109f740617953/src/main/cpp/RLBotInterface/RLBotMessages/MessageTranslation/FlatbufferTranslator.cpp#L147). I asked tarehart in discord for more info about these bad packets.

Reply:
Yeah I mean it's not necessarily a bad packet, given in the interface it's actually an optional value. The semantics of what a missing value then means is I guess up to the implementers, and I guess in this case tare & gang decided they can create packets that have game info but not ball info.